react/utils/isAdmin.ts

```diff
@@ -0,0 +1,24 @@
+import { canUseDOM } from 'exenv'
+
+/**
+ * This function checks if anything is running in the Admin environment,
+ * without relying on Render's own features.
+ */
+export function isAdmin() {
+  if (canUseDOM) {
+    const { host } = window.location
+
+    const domain = host.match(/(?<=\.)([^.]+\.[^.]+$)/)?.[0]
+
+    if (
+      domain === 'myvtex.com' &&
+      (window.location.pathname.startsWith('/admin') ||
+        // Let's consider the Admin Login page as an Admin App even though it's technically not
+        window.location.pathname.startsWith('/_v/segment/admin-login/v1/login'))
+    ) {
+      return true
+    }
+  }
+
+  return false
```

Comment:
just a suggestion to improve the readability

```suggestion
    const pathname = window.location.pathname
    const adminLoginPath = '/_v/segment/admin-login/v1/login'
    const isVtexDomain = domain === 'myvtex.com'
    const isAdminPath = pathname.startsWith('/admin') || pathname.startsWith(adminLoginPath)
    
    return isVtexDomain && isAdminPath
```

Reply:
Good one @lucasaarcoverde, I'll apply it :hugs: